base/socket.jl

```diff
@@ -282,10 +282,17 @@ mutable struct TCPSocket <: LibuvStream
         return tcp
     end
 end
-function TCPSocket()
+
+# kw arg "delay": if true, libuv delays creation of the socket fd till the first bind call
+function TCPSocket(; delay=true)
     tcp = TCPSocket(Libc.malloc(_sizeof_uv_tcp), StatusUninit)
-    err = ccall(:uv_tcp_init, Cint, (Ptr{Void}, Ptr{Void}),
-                eventloop(), tcp.handle)
+    if delay
+        err = ccall(:uv_tcp_init, Cint, (Ptr{Void}, Ptr{Void}),
+                    eventloop(), tcp.handle)
+    else
+        err = ccall(:uv_tcp_init_ex, Cint, (Ptr{Void}, Ptr{Void}, Cuint),
+                    eventloop(), tcp.handle, 2) # AF_INET is 2
+    end
```

Comment:
I think you can call the `_ex` version unconditionally, and just select between `AF_INET = 2` and `AF_UNSPEC = 0` (that's what libuv will do internally anyways)